unblob/handlers/compression/lzo.py

```diff
@@ -86,13 +95,22 @@ def calculate_chunk(
 
         logger.debug("LZO header parsed", header=header, _verbosity=3)
 
-        size_crc_header = self.cparser_be.lzo_size_crc_t(file)
-        logger.debug("CRC header parsed", header=size_crc_header, _verbosity=3)
+        while True:
```

Comment:
Just an idea to make it look less "infine-loopy":

```python
uncompressed_size = convert_int32(file.read(4), endian=Endian.BIG)
while uncompressed_size != 0:
   ...
end_offset = file.tell()
```

Reply:
I guess this is the core change to handle multi-block, and multi-block is the reason we can not tell when we need to stop.

Reply:
I think the `uncompressed_size = convert_int32(...)` bit would still be there in the loop body as well and it'd just remove the need to for the `if` block at L101. I am in favor of this change as it'd make the exit condition more obvious.

Reply:
fixed loop to make exit more explicit